<launch>
    <!--Iniciar a TF-->
    <node name="static_tf" pkg="tf" type="static_transform_publisher" args="0 0 0 0 0 0 /map /odom 100" />

    <!--Localization-->
    <node pkg="localization" type="localization_node" name="localization_node" output="screen">

    <!-- Publish scans from best pose at a max of 10 Hz -->
    <!-- Sample Motion Model-->
    <!-- odom or vel-->
    <param name="sample_motion_model" value="odom"/>

    <!--cmd_vel or odom -->
    <param name="motion_model_command_topic" value="odom"/>
    
    <!--Real world-->
    <param name="real_world_case" value="false"/>

    <!--Odometry Model alphas-->
    <param name="sample_odometry_model_alpha_1" value="0.001"/>
    <param name="sample_odometry_model_alpha_2" value="0.001"/>
    <param name="sample_odometry_model_alpha_3" value="0.001"/>
    <param name="sample_odometry_model_alpha_4" value="0.001"/>

    <!--Velocity Model alphas-->
    <param name="sample_velocity_model_alpha_1" value="0.05"/>
    <param name="sample_velocity_model_alpha_2" value="0.05"/>
    <param name="sample_velocity_model_alpha_3" value="0.05"/>
    <param name="sample_velocity_model_alpha_4" value="0.05"/>
    <param name="sample_velocity_model_alpha_5" value="0.025"/>
    <param name="sample_velocity_model_alpha_6" value="0.025"/>

    <!-- Sample Measurement Model-->

    <!-- likelihood or beam-->
    <param name="measurement_model" value="beam"/> 

    <!--Beam parameters-->
    <param name="beam_z_hit" value="0.8"/>
    <param name="beam_z_short" value="0.09"/>
    <param name="beam_z_max" value="0.06"/>
    <param name="beam_z_rand" value="0.06"/>
    <param name="beam_sigma_hit" value="0.2"/>
    <param name="beam_lambda_short" value="0.2"/>

    <!--Likelihood parameters -->
    <param name="map_max_occ_distance" value="2.0" />

    <!-- Monte carlo localization type-->
    <!--normal, injection or augmented -->
    <param name="monte_carlo_version" value="injection"/>

    <!--How many loops until resample5-->
    <param name="resample_rate" value="1"/>

    <!--Quantity Particles-->
    <param name="sample_set_size" value="1000"/>

    <!--Start spread or on robot-->
    <param name="spread_samples" value="true"/>

    <!-- injection parameters-->

    <!--How many loops to inject random particles-->
    <param name="injection_rate" value="10"/>

    <!--percentage of random particles injection-->
    <param name="random_amount_of_samplest" value="0.2"/>

    <!-- General Topics-->
    <!-- the laser topic-->
    <param name="laser_max_beams" value="180" />
    <param name="laser_scan_topic" value="scan"/>

    <param name="map_server_topic" value="map"/>

    <!--Topic to publish the particles -->
    <param name="pose_array_topic" value="pose_array"/>

    </node>

</launch>
